packages/ckeditor5-ui/src/toolbar/block/blocktoolbar.ts

```diff
@@ -458,6 +510,83 @@ export default class BlockToolbar extends Plugin {
 
 		this.buttonView.top = absoluteButtonRect.top;
 		this.buttonView.left = absoluteButtonRect.left;
+
+		this._clipButtonToViewport( this.buttonView, editableElement );
+	}
+
+	/**
+	 * Clips the button element to the viewport of the editable element.
+	 *
+	 * 	* If the button overflows the editable viewport, it is clipped to make it look like it's cut off by the editable scrollable region.
+	 * 	* If the button is fully hidden by the top of the editable, it is not clickable but still visible in the DOM.
+	 *
+	 * @param buttonView - The button view to clip.
+	 * @param editableElement - The editable element whose viewport is used for clipping.
+	 */
+	private _clipButtonToViewport(
+		buttonView: ButtonView,
+		editableElement: HTMLElement
+	) {
+		const absoluteButtonRect = buttonView.element!.getBoundingClientRect();
+		const scrollableElement = closestScrollableParent( editableElement );
+
+		if ( !scrollableElement ) {
+			return;
+		}
+
+		// Get the viewport of the scrollable element. It's not guaranteed that the scrollable element is the same as the editable element.
+		// Sometimes the editable is inside a scrollable element, so we need to get the viewport of the scrollable parent element.
+		const scrollViewportRect = scrollableElement.getBoundingClientRect();
```

Comment:
Maybe we could use our [`Rect`](https://github.com/ckeditor/ckeditor5/blob/08270aafcc6e7e4fb4f4a4d2ebbc7451f031d303/packages/ckeditor5-utils/src/dom/rect.ts#L24) wrapper.